flake.nix

```
@@ -73,6 +73,8 @@
       nixTarballs = forAllSystems ({ system, ... }:
         inputs.nix.tarballs_direct.${system}
           or "${inputs.nix.checks."${system}".binaryTarball}/nix-${inputs.nix.packages."${system}".default.version}-${system}.tar.xz");
+
+      optionalPathToDeterminateNixd = system: if builtins.elem system [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ] then "${inputs.determinate.packages.${system}.default}/bin/determinate-nixd" else null;
```

Better, but I really did mean to split out the list:
```suggestion
      supportedDeterminateNixdSystems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
      optionalPathToDeterminateNixd = system: if builtins.elem system supportedDeterminateNixdSystems then "${inputs.determinate.packages.${system}.default}/bin/determinate-nixd" else null;
```

---

Done!